src/aerovaldb/aerovaldb.py

```diff
@@ -1163,47 +1163,11 @@ async def put_by_uuid(self, obj, uuid: str):
         """
         raise NotImplementedError
 
-    @async_and_sync
-    async def acquire_lock(
-        self,
-        blocking: bool = True,
-        delay: float = 0.01,
-        max_delay: float = 0.1,
-        timeout: float | None = None,
-    ) -> bool:
-        """
-        Acquires the database lock which will prevent multiple instances
-        of aerovaldb to write to this database.
-
-        :param blocking: Whether to wait to try to acquire the lock.
-        :param delay: When blocking, starting delay as well as the delay increment (in seconds).
-        :param max_delay: When blocking the maximum delay in between attempts to acquire (in seconds).
-        :param timeout: When blocking, maximal waiting time (in seconds).
-
-        :return
-            Boolean indicating whether or not the acquisition succeeded.
-
-        Note:
-        -----
-        See also: https://fasteners.readthedocs.io/en/latest/api/inter_process/
-
-        Note:
-        -----
-        This assumes that each instance of aerovaldb uses the same lockfile
-        directory, `~/.aerovaldb/` by default, but can be set using the
-        `AVDB_LOCK_DIR` environment variable.
-        """
-        raise NotImplementedError
+    def lock(self):
+        """Acquires an exclusive advisory lock to coordinate file access
```

Comment:
please document that this returns a context-manager which has a `release` function, e.g.
```python
class AerovalDbLock(abc.ABC):
    """Abstract Lock implementation usable as context-manager"""
    @abc.abstractmethod
    def release(self):
        """release the lock manually, usually this should be done using a with-statement"""
        pass
...
def lock(self) -> AerovalDbLock:
...
class JsonLockDb(AerovalDbLock):
...
```